src/ui/server/shared-validation/phone-number/index.test.ts

```diff
@@ -0,0 +1,99 @@
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error when landline number is too long', () => {
+      const phoneNumber = '020727180101';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error when landline number is too short', () => {
+      const phoneNumber = '0207271801';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error when mobile number is too long', () => {
+      const phoneNumber = '074580452001';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error when mobile number is too short', () => {
+      const phoneNumber = '0745804520';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error when mobile number has letters in it', () => {
+      const phoneNumber = '07458O4520';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+      const expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+
+    it('should display error entering international number', () => {
+      let result = validatePhoneNumber('0144513100', fieldId, errorMessage, errors);
+      let expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+
+      result = validatePhoneNumber('+33144513100', fieldId, errorMessage, errors);
+      expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+
+      result = validatePhoneNumber('0033144513100', fieldId, errorMessage, errors);
+      expected = generateValidationErrors(fieldId, errorMessage);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('without errors', () => {
+    it('should not display error if valid landline number', () => {
+      const phoneNumber = '02072718010';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+
+      expect(result).toEqual({});
+    });
+
+    it('should not display error if valid mobile number', () => {
+      const phoneNumber = '07458045200';
+
+      const result = validatePhoneNumber(phoneNumber, fieldId, errorMessage, errors);
+
+      expect(result).toEqual({});
+    });
+  });
```

Comment:
Same comment as in E2E tests, seems there are more possibilities that we should test?

Reply:
done